src/EnemyBehaviourEC.cpp

```diff
@@ -107,6 +154,33 @@ bool EnemyBehaviourEC::timeToAttack() {
     return false;
 }
 
+Ogre::Vector3 EnemyBehaviourEC::separate() {
+
+    Ogre::Vector3 result = Ogre::Vector3(0, 0, 0);
+    int numAgents = otherTransform.size();
+    for (int i = 0; i < numAgents; i++) {
+        TransformComponent* objective = otherTransform[i];
+
+        Ogre::Vector3 myPos =
+            reinterpret_cast<TransformComponent*>(
+                this->father->getComponent("TransformComponent"))
+                ->getPosition();
```

Comment:
I also meant extracting this in two variables, also don't use `this`, this is not JavaScript.

```cpp
auto transform = reinterpret_cast<TransformComponent*>(father->getComponent("TransformComponent"));
Ogre::Vector3 myPos = transform->getPosition();
```